aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/internal/CompleteMultipartUpload.java

```diff
@@ -115,17 +115,23 @@ public UploadResult call() throws Exception {
      */
     private List<PartETag> collectPartETags() {
 
-        final List<PartETag> partETags = new ArrayList<PartETag>();
+        final List<PartETag> partETags = new ArrayList<PartETag>(eTagsBeforeResume.size());
```

Comment:
What do you think about, 
```suggestion
        final List<PartETag> partETags = new ArrayList<PartETag>(eTagsBeforeResume.size() +  futures.size());
```